src/game/client/draw/drawable.cpp

```diff
@@ -3069,3 +3083,211 @@ void Drawable::Apply_Physics_Xform(Matrix3D *mtx)
         }
     }
 }
+
+void Drawable::Set_Fully_Obscured_By_Shroud(bool fully_obscured)
+{
+    for (DrawModule **i = Get_Draw_Modules(); *i != nullptr; i++) {
+        (*i)->Set_Fully_Obscured_By_Shroud(fully_obscured);
+    }
+
+    m_fullyObscuredByShroud = fully_obscured;
+}
+
+void Drawable::Update_Drawable()
+{
+    unsigned int frame = g_theGameLogic->Get_Frame();
+    Object *object = Get_Object();
+
+    for (ClientUpdateModule **i = Get_Client_Update_Modules(); i != nullptr && *i != nullptr; i++) {
+        (*i)->Client_Update();
+    }
+
+    if (m_fadingMode != FADING_MODE_OFF) {
+        Set_Opacity(
+            (float)(m_fadingMode == FADING_MODE_OUT ? m_curFadeFrame : m_timeToFade - m_curFadeFrame) / (float)m_timeToFade);
+        m_curFadeFrame++;
+
+        if (m_curFadeFrame > m_timeToFade) {
+            m_fadingMode = FADING_MODE_OFF;
+        }
+    }
+
+    if (Get_Terrain_Decal() == TERRAIN_DECAL_8) {
+        m_terrainDecalOpacity = 0.0f;
+    } else {
+        DrawModule **modules = Get_Draw_Modules();
+
+        if (*modules != nullptr) {
+            if (m_terrainDecalFadeTarget2 != 0.0f) {
+                (*modules)->Set_Terrain_Decal_Opacity(m_terrainDecalOpacity);
+            }
+
+            if (m_terrainDecalFadeTarget2 >= 0.0f || m_terrainDecalOpacity > 0.0f) {
+                if (m_terrainDecalFadeTarget2 > 0.0f && m_terrainDecalOpacity >= 1.0f) {
+                    m_terrainDecalOpacity = 1.0f;
+                    m_terrainDecalFadeTarget2 = 0.0f;
+                    (*modules)->Set_Terrain_Decal_Opacity(m_terrainDecalOpacity);
+                }
+            } else {
+                m_terrainDecalFadeTarget2 = 0.0f;
+                m_terrainDecalOpacity = 0.0f;
+                Set_Terrain_Decal(TERRAIN_DECAL_8);
+            }
+        }
+    }
+
+    if (m_expirationDate && frame >= m_expirationDate) {
+        captainslog_dbgassert(object == nullptr, "Drawables with Objects should not have expiration dates!");
+        g_theGameClient->Destroy_Drawable(this);
```

Comment:
early return was here, thus eliminating the else below

Reply:
fixed.